qt_ui/windows/QLiberationWindow.py

```diff
@@ -367,6 +371,32 @@ def onGameGenerated(self, game: Game):
         self.game = game
         GameUpdateSignal.get_instance().game_loaded.emit(self.game)
 
+    def onEndGame(self, state: TurnState):
+        result = None
+        if state == TurnState.WIN:
+            result = QMessageBox.information(
+                QApplication.focusWidget(),
+                "Victory!",
+                "You have won the campaign, do you wish to start a new one?",
+                QMessageBox.Yes,
+                QMessageBox.No,
+            )
+        elif state == TurnState.LOSS:
+            result = QMessageBox.information(
+                QApplication.focusWidget(),
+                "Defeat!",
+                "You have lost the campaign, do you wish to start a new one?",
+                QMessageBox.Yes,
+                QMessageBox.No,
+            )
+        for window in QApplication.topLevelWidgets():
+            if window is not self:
+                window.close()
+        sleep(0.5)
+        GameUpdateSignal.get_instance().updateGame(None)
```

Comment:
This prevents the user from clicking on the map, opening CP's and what not, to possibly revert the state of the game through the cheat button. This also prevents the user from buying units or transferring them, which would be useless if the campaign has ended. It also removes all packages and flights from the ATO panel which is what we want, or at least that's what I assume.
One small issue I noticed is that it's possible one of the flight-plans might get selected on the map, allowing you to move the markers around even though the flight doesn't exist anymore. The lines of the flight-plan stay drawn the same way, it's literally only the pin-marker that becomes movable. Now as far as I know, this shouldn't cause any exceptions so I think leaving it as it is is a trade-off we should consider.